src/radical/pilot/agent/resource_manager/pbspro.py

```diff
@@ -20,35 +19,35 @@ class PBSPro(ResourceManager):
     #
     def _init_from_scratch(self, rm_info: RMInfo) -> RMInfo:
 
-        # TODO: $NCPUS?!?! = 1 on archer
+        try:
+            vnodes, rm_info.cores_per_node = self._parse_pbspro_vnodes()
+            nodes = [(node, rm_info.cores_per_node) for node in vnodes]
 
-        cpn = os.environ.get('NUM_PPN') or os.environ.get('SAGA_PPN')
-        if not cpn:
-            raise RuntimeError('$NUM_PPN and $SAGA_PPN not set!')
+        except (IndexError, ValueError) as exc:
+            self._log.debug_2('exec_vnodes not detected')
 
-        pbspro_vnodes = self._parse_pbspro_vnodes()
+            if not rm_info.cores_per_node or 'PBS_NODEFILE' not in os.environ:
+                raise RuntimeError('resource configuration unknown') from exc
```

Comment:
Same here: not sure the error message fits the error.  I would not know what to look for reading that message (or would look in the wrong place, i.e., the resource configs).

Reply:
extended